src/targets/gpu/device/include/migraphx/gpu/device/float_equal.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_GPU_DEVICE_FLOAT_EQUAL_HPP
+#define MIGRAPHX_GUARD_RTGLIB_GPU_DEVICE_FLOAT_EQUAL_HPP
+
+#include <hip/hip_runtime.h>
+#include <migraphx/half.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/tensor_view.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+template <class... Ts>
+using common_type = typename std::common_type<Ts...>::type;
+
+struct float_equal_fn
+{
+    template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
```

Comment:
I am not sure if `is_floating_point` will return true for `__fp16`. I think we should just define a `is_floating_point` trait in the types.hpp header. This will avoid the dependency on <migraphx/half.hpp>.

Reply:
@pfultz2 I changed it to a function and added a trait in the types.hpp for the __fp16, but looks like it does not work. More specifically, after I changed it to a function, it begin not working. Can you help check what is wrong with that?

Reply:
Never mind. I was able to fix it and it works now.